code.py

```diff
@@ -53,12 +56,33 @@ def nextLine(self, pose):
 
 
 	# Function D in paper: returns dist b/w oldpose and next line
-	def nextDist(self, pose, line):
+	def nextDist(self, pose, line, index_bot):
+		x0 = pose[0]
+		x1 = pose[1]
+		x1 = line[0][0]
+		y1 = line[0][1]
+		x2 = line[1][0]
+		y2 = line[1][1]
+		
+		d1 = math.sqrt((x1 - x0)**2 + (y1 - y0)**2)
+		d2 = math.sqrt((x2 - x0)**2 + (y2 - y0)**2)
+		
+		if d1 < d2:
+			self.pose[index_bot] = [x2, y2]
+		else:
+			self.pose[index_bot] = [x1, y1]
+		dist = min(d1,d2)
 		return dist
 
 
 	# Heuristics func in paper: Return distance/cost of a single input group
-	def groupDist(self, group):
```

Comment:
This is the heuristic() function in the paper. It takes a group as an input, and orders the lines in the group, not an individual.

Reply:
oh okay. Isn't the group referring to one of the solution because it returns cost of that. I will go through the paper as well

Reply:
Here the cost will return the total distance traveled by all bots in that particular solution in the population.

Reply:
Groups refers to list of lines for the i-th robot. For ex. group_1 is a set of line that robot 1 prints. Hence, you find robot 1's cost.

Reply:
Oh okay. I get what needs to be done. 
But bhaiya, I have a doubt. 
Getting individual cost of each robot in the solution and then trying to make them all as close to each other as possible.
And Getting the total cost of the solution and minimizing it as a whole.

Won't they both result in the same thing. I get the point, will work on it as well.
I am really sorry if I sound rude or anything.

Reply:
In this, c[i] will store the array of costs of the respective bots, and the max of them will be returned. Please tell me if this is correct or do I need to make changes.